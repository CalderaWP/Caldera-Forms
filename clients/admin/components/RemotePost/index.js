import {
	RemotePost
} from "./RemotePost";

import postPropTypes from './propTypes';

export default {
	RemotePost,
	postPropTypes
}