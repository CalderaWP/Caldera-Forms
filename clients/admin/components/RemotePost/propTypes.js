import PropTypes from 'prop-types'

/**
 * Prop type definitions for the RemotePost component
 *
 * @type {{form: shim, formId: (boolean|*), onFormUpdate: (boolean|*)}}
 */
export const postPropTypes = {
	post: PropTypes.shape({
		id: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.number
		]).isRequired,
		title: PropTypes.shape({
			rendered: PropTypes.string.isRequired,
		}),
		content: PropTypes.shape({
			rendered: PropTypes.string.isRequired,
		}),
	}).isRequired
};

