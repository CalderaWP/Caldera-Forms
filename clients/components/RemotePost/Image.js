import React from 'react';

export default function (props) {
    return (
        <img
            src={props.src}
            className={props.className}
            width={props.width}
            height={props.height}
            alt={props.alt}
            style={ props.style ? props.style :
                {
                    width: '100%',
                    height: 'auto'
                }
            }
            lazy="true"
        />
    )
}

