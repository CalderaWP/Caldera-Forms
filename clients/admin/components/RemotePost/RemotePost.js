import React from 'react';
import {postPropTypes} from "./propTypes";
import EmbedContainer from 'react-oembed-container';
import PropTypes from 'prop-types'

/**
 * The main container component for the RemotePost
 *
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export const RemotePost = (props) => {
	const {post} = props;

	const Content = () => {
		if (props.showExcerpt) {
			return (<div dangerouslySetInnerHTML={{__html: post.excerpt.rendered}}/>);

		}
		return (
			<div dangerouslySetInnerHTML={{__html: post.content.rendered}}/>
		);

	}
	return (
		<EmbedContainer
			markup={post.content.rendered}
			className={props.className}

		>
			<article
				id={`post-${post.id}`}
			>
				<h2>
					{post.title.rendered}
				</h2>
				{Content()}
				<a
					href={props.post.href}
					target="_blank"
					className={props.buttonClassName}

				>
					{props.readMore}
				</a>
			</article>

		</EmbedContainer>
	);
};

RemotePost.propTypes = {
	...postPropTypes,
	className: PropTypes.string,
	showExcerpt: PropTypes.bool,
	readMore: PropTypes.string

};

RemotePost.defaultProps = {
	showExcerpt: true,
	readMore: 'Learn More',
	buttonClassName: 'btn btn-green'

}
