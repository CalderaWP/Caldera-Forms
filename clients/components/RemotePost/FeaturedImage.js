import React from 'react';
import axios from 'axios';
import Image from './Image';
import url from '../functions/url';
import {cacheAdapterEnhancer} from 'axios-extensions';

export class FeaturedImage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            loaded: true,
            alt_text: '',
            width: 1484,
            height: 592,
            source_url: 'https://calderaforms.com/wp-content/uploads/2017/12/forms_copy.jpg',

            markup: '',
        };

    }

    componentDidMount() {
        axios({
            method: 'get',
            url: `${this.props.apiRoot}/wp/v2/media`,
            params: {
                parent: this.props.post.id
            },
            adapter: cacheAdapterEnhancer(axios.defaults.adapter, true)
        }).then((response) => {
            const newImage = response.data[0];
            if (newImage && newImage.hasOwnProperty('alt_text')) {
                this.setState({
                    alt_text: newImage.alt_text,
                    source_url: newImage.source_url,
                    width: newImage.media_details.width,
                    height: newImage.media_details.height,
                    markup: newImage.hasOwnProperty('description') ? newImage.description.rendered : ''
                });
            }

        })
    }



    render() {
        return (
            <div>
                <a
                    href={url(this.props.lastParams, this.props.post.link)}
                >
                    <Image
                        className=""
                        alt={this.state.alt_text}
                        width={this.state.width}
                        height={this.state.height}
                        src={this.state.source_url}
                    />
                </a>

            </div>

        )

    }


}