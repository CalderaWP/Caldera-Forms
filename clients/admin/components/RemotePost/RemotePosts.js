import React from 'react';
//import type {wpPost} from "../../types/types/wpPost";
import Grid from 'react-css-grid';
import {RemotePost} from "./RemotePost";

export const RemotePosts = (props) => {
	const {posts} = props;
	const readMore = props.readMore ? props.readMore : 'Learn More';
	return (
		<Grid>
			{
				posts.map(function (post) {
					return (
						<RemotePost
							post={post}
							key={post.id}
							readMore={readMore}
						/>
					)
				})
			}
		</Grid>
	)
};