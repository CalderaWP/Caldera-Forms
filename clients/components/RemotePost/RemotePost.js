import React from 'react';
import {FeaturedImage} from "./FeaturedImage";
import {Grid, Row, Col} from 'react-bootstrap';
import url from '../functions/url';
import PostExcerpt from './PostExcerpt';
import PostTitle from './PostTitle';
export class RemotePost extends React.Component {





    render() {
        const {post,lastParams,apiRoot} = this.props;
        return (
            <Grid>
                <article
                    id={`post-${post.id}`}
                    className={`post-${post.id} row not-box hentry`}
                >
                    <Row>
                        <Col
                            xs={12}
                        >

                        </Col>
                    </Row>

                    <Row>
                        <Col
                            xs={12}
                            sm={3}
                            md={4}
                        >
                            <FeaturedImage
                                apiRoot={apiRoot}
                                post={post}
                                lastParams={lastParams}
                            />
                        </Col>
                        <Col
                            xs={12}
                            sm={12}
                            md={8}
                        >
                            <PostExcerpt post={post}/>
                        </Col>
                        <Col
                            xs={12}
                        >
                            <a
                                href={url(lastParams, post.link)}
                                target="_blank"
                                className="btn btn-green btn-block btn-readmore"
                                rel="bookmark"
                            >
                                Read More
                            </a>
                        </Col>
                    </Row>


                    <footer className="entry-footer">
                    </footer>


                </article>
            </Grid>


        )
    }

}