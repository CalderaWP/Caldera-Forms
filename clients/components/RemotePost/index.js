import {RemotePost} from "./RemotePost";
import {FeaturedImage} from "./FeaturedImage";
import Image from "./Image";
import PostExcerpt from './PostExcerpt'
import PostTitle from './PostTitle'
import RemoteImage from './RemoteImage'
export {
    RemotePost,
    FeaturedImage,
    Image,
    PostExcerpt,
    PostTitle,
    RemoteImage
}

