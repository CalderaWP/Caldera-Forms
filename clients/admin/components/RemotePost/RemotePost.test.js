import React from 'react';
import renderer from "react-test-renderer";
import {mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {RemotePost} from "./RemotePost";
Enzyme.configure({adapter: new Adapter()});


const post = {
	id: 42,
	title : {
		rendered: 'Hello To Roy'
	},
	excerpt: {
		rendered: '<p>Hi R| Read more...</p>'
	},
	content: {
		rendered: '<p>Hi Roy</p>'
	},
	href: 'https://calderaforms.com/hi-roy'
};
describe( 'Remote post  Component', () => {

	it( 'Matches snapshot', () => {
		const component = renderer.create(
			<RemotePost
				post={post}

			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});

	it( 'Matches snapshot with class names', () => {
		const component = renderer.create(
			<RemotePost
				post={post}
				className={'remote-post-for-help'}
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});

	it( 'Matches snapshot with read more text', () => {
		const component = renderer.create(
			<RemotePost
				post={post}
				className={'remote-post-for-help'}
				readMore={'Hi Roy - Read More'}
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});

	it( 'Matches snapshot with button class name', () => {
		const component = renderer.create(
			<RemotePost
				post={post}
				className={'remote-post-for-help'}
				buttonClassName={'roys'}
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});
});
