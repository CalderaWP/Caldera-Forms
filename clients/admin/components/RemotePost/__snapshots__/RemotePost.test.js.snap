// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remote post  Component Matches snapshot 1`] = `
<div
  className={null}
>
  <article
    id="post-42"
  >
    <h2>
      Hello To Roy
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hi R| Read more...</p>",
        }
      }
    />
    <a
      className="btn btn-green"
      href="https://calderaforms.com/hi-roy"
      target="_blank"
    />
  </article>
</div>
`;

exports[`Remote post  Component Matches snapshot with button class name 1`] = `
<div
  className="remote-post-for-help"
>
  <article
    id="post-42"
  >
    <h2>
      Hello To Roy
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hi R| Read more...</p>",
        }
      }
    />
    <a
      className="roys"
      href="https://calderaforms.com/hi-roy"
      target="_blank"
    />
  </article>
</div>
`;

exports[`Remote post  Component Matches snapshot with class names 1`] = `
<div
  className="remote-post-for-help"
>
  <article
    id="post-42"
  >
    <h2>
      Hello To Roy
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hi R| Read more...</p>",
        }
      }
    />
    <a
      className="btn btn-green"
      href="https://calderaforms.com/hi-roy"
      target="_blank"
    />
  </article>
</div>
`;

exports[`Remote post  Component Matches snapshot with read more text 1`] = `
<div
  className="remote-post-for-help"
>
  <article
    id="post-42"
  >
    <h2>
      Hello To Roy
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Hi R| Read more...</p>",
        }
      }
    />
    <a
      className="btn btn-green"
      href="https://calderaforms.com/hi-roy"
      target="_blank"
    />
  </article>
</div>
`;
